app/facades/cocktail_facade.rb

```diff
@@ -0,0 +1,56 @@
+class CocktailFacade
+  def self.retrieve_cocktail(cocktail_id)
+    response = CocktailService.get_cocktail_details(cocktail_id)
+
+    details = response[:drinks].first
+
+    ingredients = get_ingredients(details)
+    measurements = get_measurements(details)
+
+    recipe = measurements.zip(ingredients).map do |i|
+      i.join if i != ["", ""]
+    end.compact
+
+    rating = if Rating.find_by(cocktail_id: cocktail_id).present?
+      Rating.find_by(cocktail_id: cocktail_id).stars
+    else
+      0
+    end
+
+    cocktail = {
+      # id: details[:idDrink],
+      name: details[:strDrink],
+      thumbnail: details[:strDrinkThumb],
+      glass: details[:strGlass],
+      recipe: recipe,
+      instructions: details[:strInstructions],
+      rating: rating
+    }
+
+    # CocktailDetails.new(cocktail)
+  end
+
+  def self.get_ingredients(cocktail_data)
```

Comment:
this can be refactored similarly to the get_measurements method
get_ingredients and get_measurements are identical aside from the "ingredient" and the "measurement" that are being passed as the argument to the .include? it is very clear which specific thing we are getting by having two methods, but I'm wondering if there is a way to dry up this code that still communicated the clarity that we get by having two separate methods.